flixel/input/gamepad/FlxGamepad.hx

```diff
@@ -719,13 +852,55 @@ class FlxGamepadAnalogStick
 	public var x(default, null):Int;
 	public var y(default, null):Int;
 	
-	public function new(x:Int, y:Int)
+	//these values let the analog stick to send digital inputs to, say, the dpad
+	public var rawUp(default, null):Int = -1;
+	public var rawDown(default, null):Int = -1;
+	public var rawLeft(default, null):Int = -1;
+	public var rawRight(default, null):Int = -1;
+	
+	//the value the dpad must be above before digital inputs are sent
+	public var digitalThreshold(default, null):Float = 0.5;
+	
+	//when analog inputs are received, how to process them digitally
+	public var mode(default, null):FlxAnalogToDigitalMode = OnlyAnalog;
+	
+	public function new(x:Int, y:Int, ?settings:FlxGamepadAnalogStickSettings)
 	{
 		this.x = x;
 		this.y = y;
+		if (settings != null)
+		{
+			mode     = (settings.mode  != null ? settings.mode  : OnlyAnalog);
+			rawUp    = (settings.up    != null ? settings.up    : -1);
+			rawDown  = (settings.down  != null ? settings.down  : -1);
+			rawLeft  = (settings.left  != null ? settings.left  : -1);
+			rawRight = (settings.right != null ? settings.right : -1);
+			digitalThreshold = (settings.threshold != null ? settings.threshold : -1);
+		}
+	}
+	
+	public function toString():String
+	{
+		return("stick(" + x + "," + y + ",(" + rawUp + "," + rawDown + "," + rawLeft + "," + rawRight + ") @" + digitalThreshold + ":" + mode+")");
```

Comment:
Might make sense to use [Flixel's standard toString() logic](https://github.com/HaxeFlixel/flixel/blob/dev/flixel/FlxObject.hx#L1061-L1070)?